src/main/groovy/org/edx/jenkins/dsl/AnalyticsConstants.groovy

```diff
@@ -216,5 +216,29 @@ This text may reference other parameters in the task as shell variables, e.g.  $
             }
         }
     }
+    public static def slack_publisher = { allVars ->
+        return {
+            it /
+                publishers /
+                'jenkins.plugins.slack.SlackNotifier' {
```

Comment:
Are we unable to use `slackNotifier` directly in the publisher context ?
http://jenkins-new.analytics.edx.org/plugin/job-dsl/api-viewer/index.html#method/javaposse.jobdsl.dsl.helpers.publisher.PublisherContext.slackNotifier
I think this might work ? unless already tried and it does not.
```
publishers {
    slackNotifier {
        room('$SLACK_NOTIFICATION_CHANNEL')
        ....
    }
}

Reply:
Ok let me try this way

Reply:
Yes that is working, thanks
So one suggestion needed should I put this code in AnalyticsConstants.groovy or inside the job DSL code like here in testing PR: https://github.com/edx/jenkins-job-dsl/pull/1358/files
@HassanJaveed84

Reply:
If we expect to be using the notification in future for other jobs, we should put it in the constants file and make it re-usable.

Reply:
Ok I will prefer putting it in AnalyticsConstants file because there are chances of using slack notifications in future jobs.

Reply:
Hey! updated as per our discussion @HassanJaveed84

Reply:
I notice that you're passing `allVars` to the `slack_publisher` method. I'd change

`room('$SLACK_NOTIFICATION_CHANNEL')` -> `room(allVars.get('SLACK_NOTIFICATION_CHANNEL'))`
  `customMessageFailure('$FAILURE_MESSAGE')` -> `customMessageFailure(allVars.get('FAILURE_MESSAGE'))`
  
  and then we can remove the parameters defined at the top.

Reply:
sure, fixed it